feature/experimental/gribi/otg_tests/route_removal_during_failover_test/route_removal_during_failover_test.go

```diff
@@ -0,0 +1,703 @@
+	// configure DUT port#1 - source port.
+	configureSubinterfaceDUT(d, dp1, 0, 0, dutPort1.IPv4)
+	configureInterfaceDUT(t, dp1, d, "src")
+	configureATE(top, ap1, "src", 0, dutPort1.IPv4, atePort1.IPv4, atePort1.MAC)
+	pushConfig(t, dut, dp1, d)
+	dp2 := dut.Port(t, "port2")
+	ap2 := ate.Port(t, "port2")
+	top.Ports().Add().SetName(ap2.ID())
+	// Configure 64 subinterfaces on DUT-ATE- PORT#2.
+	subIntfIPs := generateSubIntfPair(t, dut, dp2, ate, ap2, top, d)
+	if *deviations.ExplicitInterfaceInDefaultVRF {
+		fptest.AssignToNetworkInstance(t, dut, dp1.Name(), *deviations.DefaultNetworkInstance, 0)
+	}
+	if *deviations.ExplicitPortSpeed {
+		fptest.SetPortSpeed(t, dp1)
+		fptest.SetPortSpeed(t, dp2)
+	}
+	if *deviations.ExplicitGRIBIUnderNetworkInstance {
+		fptest.EnableGRIBIUnderNetworkInstance(t, dut, *deviations.DefaultNetworkInstance)
+	}
+
+	ate.OTG().PushConfig(t, top)
+	ate.OTG().StartProtocols(t)
+
+	dutPortName := dut.Port(t, "port1").Name()
+	sysConfigTime := gnmi.Get(t, dut, gnmi.OC().Interface(dutPortName).LastChange().State())
+
+	// Connect gRIBI client to DUT referred to as gRIBI - using PRESERVE persistence and
+	// SINGLE_PRIMARY mode, with FIB ACK requested. Specify gRIBI as the leader.
+	client := fluent.NewClient()
+	client.Connection().WithStub(gribic).WithPersistence().WithInitialElectionID(1, 0).
+		WithFIBACK().WithRedundancyMode(fluent.ElectedPrimaryClient)
+	client.Start(ctx, t)
+	defer client.Stop(t)
+
+	defer func() {
+		// Flush all entries after test.
+		if err := gribi.FlushAll(client); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	client.StartSending(ctx, t)
+	if err := awaitTimeout(ctx, client, t, time.Minute); err != nil {
+		t.Fatalf("Await got error during session negotiation for clientA: %v", err)
+	}
+	eID := gribi.BecomeLeader(t, client)
+
+	t.Log("Flush existing gRIBI routes before test.")
+	if err := gribi.FlushAll(client); err != nil {
+		t.Fatal(err)
+	}
+
+	args := &testArgs{
+		ctx:        ctx,
+		client:     client,
+		dut:        dut,
+		ate:        ate,
+		top:        top,
+		electionID: eID,
+	}
+
+	virtualIPs := createIPv4Entries(t, ipBlock1)
+
+	t.Log("Inject routes from ipBlock1 in default VRF with NHGID: #1.")
+	pushDefaultEntries(t, args, subIntfIPs, virtualIPs)
+
+	gr, err := args.client.Get().
+		WithNetworkInstance(*deviations.DefaultNetworkInstance).
+		WithAFT(fluent.IPv4).
+		Send()
+	if err != nil {
+		t.Fatalf("Got unexpected error from get, got: %v", err)
+	}
+
+	for _, r := range gr.GetEntry() {
+		entry := r.GetIpv4().Prefix
+		if got := r.GetFibStatus().String(); got != "PROGRAMMED" {
+			t.Fatalf("gRIBI entry %v is not programmed in FIB, got:%v, want: PROGRAMMED", entry, got)
+		}
+	}
+
+	// Send traffic from ATE port-1 to prefixes in ipBlock1 and ensure traffic
+	// flows 100% and reaches ATE port-2.
+	waitOTGARPEntry(t)
+	dstMac := gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Interface(atePort1.Name+".Eth").Ipv4Neighbor(dutPort1.IPv4).LinkLayerAddress().State())
```

Comment:
Same issue with the interface name as in the waitOTGARPEntry function.

Reply:
Same as with the waitOTGARPEntry function, the destination mac will only be checked on the transmitting port...atePort1.